src/Shared/Microsoft.AspNetCore.Razor.Test.Common/IntegrationTestAttributes.cs

```diff
@@ -36,6 +36,15 @@ public override IEnumerable<IXunitTestCase> Discover(ITestFrameworkDiscoveryOpti
 internal class IntegrationTheoryDiscoverer(IMessageSink diagnosticMessageSink)
     : TheoryDiscoverer(diagnosticMessageSink)
 {
+    public override IEnumerable<IXunitTestCase> Discover(ITestFrameworkDiscoveryOptions discoveryOptions, ITestMethod testMethod, IAttributeInfo theoryAttribute)
+    {
+        // We have to force pre-enumeration of theories for this discoverer to work correctly. Normally its true in VS,
+        // but false in command line/CI. Since we're injecting "fake" data rows, we rely on it everywhere. Without this
+        // set to true, the method below that we override doesn't get called.
+        discoveryOptions.SetValue("xunit.discovery.PreEnumerateTheories", true);
```

Comment:
FYI @chsienki 

Found a problem with the custom theory discoverer that I copied for the formatting tests. The tests that use it weren't actually running against different code-gen in CI.

eg, this should have had 4 cases:
![image](https://github.com/user-attachments/assets/0877eb4d-7528-4df4-bc17-dcce0f5b80ac)

Reply:
After:
![image](https://github.com/user-attachments/assets/f9d47572-bf54-4332-a865-7959206ef68d)